### PR TITLE
Merge basic tutorial from docs into basic hands on

### LINKED ISSUE #238: basics: verdi process list output unexpected?
The output of `verdi process list -S finished` is not as described [here](https://aiida-tutorials.readthedocs.io/en/2020-intro-week/pages/2020_Intro_Week/sections/basics.html#processes).

In particular, there are 171 results, not 6.

Furthermore, the result of `verdi process list -S finished` and `verdi process list -a` is identical (all processes have "Finished" as the process state)

### LINKED ISSUE #239: 🐛 FIX: Structure visualization not working reliably
For me, the line `verdi data structure show --format ase <IDENTIFIER>` only "kind of" works - it opens the ASE window locally, but it's not really interactive (e.g. I cannot click & drag the molecule).

Launching xrysden straight up fails with
```
    (file "/usr/share/xcrysden/Tcl/xcInit.tcl" line 609)
X Error of failed request: BadWindow (invalid Window parameter)
    Major opcode of failed request: 12
    ResourceID in failed request: 0xffffffff
    Serial number of failed request: 776
```

Perhaps this has to do with my local window manager?

### LINKED ISSUE #240: basics: section on codes unclear
The [section on codes and computers](https://aiida-tutorials.readthedocs.io/en/2020-intro-week/pages/2020_Intro_Week/sections/basics.html#codes-and-computers) asks the user to list the codes available, but there are none (note that the "First taste" section was left out!).

The text clearly seems to expect a result (e.g. `Now, among the entries of the output you should also find the code just shown.`).
Given that the users only start setting up their own computers and codes in the following section 3, I wonder whether this subsection 2.1.5.3 can be deleted entirely?

### LINKED ISSUE #242: basics: takes >1h before running your first calculation
This may be something for the 2021 edition of the tutorial, but after going through the "basics" section, I have to say that this is a very long way for anyone before they ever get to run any calculations themselves, with lots of details, e.g. on kpoints, upfdata etc. that won't matter to many folks.

Admittedly, this is how we started the May tutorial last year that I organized (mea culpa), but we realized this issue and the "first taste" section (as e.g. [here](https://aiida-tutorials.readthedocs.io/en/latest/pages/2019_SINTEF/sections/first_taste.html)) was created for exactly this reason and used in a number of later tutorials
In a virtual tutorial, where people are less committed to stay, getting to the point quickly is even more important than in a physical one.

An alternative route than the "First taste" would start with the "Basic Tutorial" from the documentation, and then add bits and pieces where necessary.

It's too late to change this now, but let's keep this in mind next time @giovannipizzi @mbercx @chrisjsewell @csadorf @sphuber

### PR DESCRIPTION
**Pull request by mbercx**
Fixes #238
Fixes #242

This PR is a rather large revamp of the AiiDA basics section of the tutorial, which now starts from the "basic tutorial" from the new AiiDA documentation. In order to still introduce the data types required for the rest of the tutorial, the tutorial continues along the lines of the previous version, but the sections have been restructured significantly, as splitting them up in verdi CLI and verdi shell did not make any sense after we already combine the two in the first part of the section.

My idea was to sort the content by data type and then discuss the the PwCalculation process, its inputs and then outputs in that order. I've tried my best to avoid too much repetition, or added words like "remember that..." to indicate that they've seen this previously. I'm still going to have another pass at this tomorrow with a clearer head, but feel free to already have a look and give suggestions.

Some notes:
* I removed the provenance graph with the inputs only. I didn’t really understand the reason of showing both here, especially straight after each other.
* I moved the groups section to the appendix. This might be more at home in the managing data and querying hands on.
* I've also moved some sections that explore the provenance to the appendix to keep the hands on from getting *too* big.

Some open issues/questions:
* The basic tutorial from the documentation introduces the computer with label `tutor` . In the "running computations" hands on, a computer with label `localhost`  is configured. Should we only configure one computer, or is it fine to keep both so they see multiple computers when they execute `verdi computer list`?
* Currently the hands on is not entirely consistent on the use of e.g. `<IDENTIFIER>` versus actually providing the PK/UUID in the CLI command/code snippet. 
* It might be better to move the outputs section to the "running computations" hands-on, where they can explore the outputs of the calculations they just ran.

### CODE REVIEW

File: docs/pages/2020_Intro_Week/appendices/exploring_provenance.rst [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+******************************
+Exploring the provenance graph
+******************************
+
+Accessing node inputs and outputs
+=================================
+
+Load again the calculation node from the :ref:`2020_virtual_aiidagraph` section:
+
+.. code:: python
```
Comment (chrisjsewell) [inline docs/pages/2020_Intro_Week/appendices/exploring_provenance.rst]: I think all of these should be ipython code-block's, with appropriate prefixes

File: docs/pages/2020_Intro_Week/appendices/exploring_provenance.rst [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+    calc.get_outgoing(link_label_filter='output%').all()
+
+The provenance browser
+======================
+
+While the ``verdi`` CLI provides full access to the data underlying the provenance graph, a more intuitive tool for browsing AiiDA graphs is the interactive provenance browser available on `Materials Cloud <https://www.materialscloud.org>`__.
+
+In order to use it, we first need to start the `AiiDA REST API <https://aiida-core.readthedocs.io/en/latest/restapi/index.html>`_:
```
Comment (chrisjsewell) [inline docs/pages/2020_Intro_Week/appendices/exploring_provenance.rst]:
```suggestion
In order to use it, we first need to start the :ref:`AiiDA REST API <reference:rest-api>`:
```

File: docs/pages/2020_Intro_Week/appendices/groups.rst [hunk 3]
```diff
@@ -0,0 +1,29 @@
+.. _2020_virtual_intro:groups:
+
+*********************
+AiiDA groups of nodes
+*********************
+
+In AiiDA, calculations (and more generally nodes) can be organized in groups, which are particularly useful to organise your data, and assign a set of calculations or data to a common subproject.
```
Comment (chrisjsewell) [inline docs/pages/2020_Intro_Week/appendices/groups.rst:7]: add reference to `` :ref:`aiida:how-to:data:organize` ``

**Review by chrisjsewell — CHANGES_REQUESTED** [review]

**Comment by chrisjsewell** [comment]
thanks @mbercx, I've also linked a number of issues that I think this should close, but maybe have a check if this is the case

File: docs/pages/2020_Intro_Week/sections/basics.rst [hunk 4]
```diff
@@ -269,35 +763,36 @@ The first three states are 'active' states, meaning the process is not done yet,
 Once a process is in a terminal state, it will never become active again.
 The `official documentation <https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/processes/concepts.html#process-state>`_ contains more details on process states.
```
[the remark this answers is not in the document]
Comment (mbercx) [reply docs/pages/2020_Intro_Week/sections/basics.rst:768]: I'm not sure if that's a good idea. The basics tutorial is already quite long as it is. Maybe we can address #218 during the running computations section, where they can pause the calculation they are running?

File: docs/pages/2020_Intro_Week/sections/basics.rst [hunk 5]
```diff
@@ -332,581 +827,533 @@ Producing the output:
...
+You should be able to rotate the view with the right mouse button.
 
-In this section we will use an interactive IPython environment with all the
-basic AiiDA classes already loaded. We propose two realizations of such a
-tool. The first consists of a special IPython shell where all the AiiDA
-classes, methods and functions are accessible. Type in the terminal
+.. note::
 
-.. code:: bash
+    If you receive some errors, make sure your X-forwarding settings have been set up correctly.
```
[the remark this answers is not in the document]
Comment (mbercx) [reply docs/pages/2020_Intro_Week/sections/basics.rst]: I've linked to the setup section of the tutorial instead.

**Comment by mbercx** [comment]
Thanks @chrisjsewell for the review! I've applied all of your suggestions, save for some unresolved conversations.

@ltalirz or @giovannipizzi: Considering the large number of changes, it would be great if you could have a look at the whole section and see if you agree with the content and structure.

**Comment by chrisjsewell** [comment]
> I've applied all of your suggestions

You still haven't figured out how to run pre-commit though have you 😜

**Comment by ltalirz** [comment]
@mbercx Thanks a lot for all this work!

I will first need to go through the plugins/code setup howtos before going through this section in-depth, but let me give some initial feedback already now:

 * I think the structure you chose it not a bad one, but logically I would prefer to keep the "AiiDA basics" section to just the basic tutorial, name the second section "Quantum ESPRESSO example", and have it contain both the data analysis and calc execution aspects.

   You may worry a bit about the relative length of both sections in this case. I would argue that this is not an issue.
   The first section is the most important one, and people will spend a lot of time getting stuck at seemingly simple, minor issues.
     Plus, people who finish quickly can just go ahead with the second one.

  Note: You might want to add a comment in the "AiiDA basics" section to mention that it is taken from the basic aiida tutorial (even if just for us to remember in the future)


 * In the QE example, don't be shy to remove things that you consider duplication. 
   E.g. explanations on graph generation, explanation of identifiers etc.

> I moved the groups section to the appendix. This might be more at home in the managing data and querying hands on.

I agree that this information can be moved to data/querying, but in that case please move it there.
Groups are an important tool and deserve to be mentioned outside appendices.

> I've also moved some sections that explore the provenance to the appendix to keep the hands on from getting too big.

One thing I am missing is the interactive provenance exploration using the Materials Cloud provenance explorer.
I think this should be reinstated (wherever it makes sense).

> The basic tutorial from the documentation introduces the computer with label tutor . In the "running computations" hands on, a computer with label localhost is configured. Should we only configure one computer, or is it fine to keep both so they see multiple computers when they execute verdi computer list?

Ideally it would be the same computer (unless we can think of an educational reason to create a second one. It might confuse participants ("do I need to create a new computer for every code?")). If we don't manage to change it, it's ok to leave it.

> Currently the hands on is not entirely consistent on the use of e.g. <IDENTIFIER> versus actually providing the PK/UUID in the CLI command/code snippet.

I don't consider it mandatory to be 100% consistent on this.
I'll keep an eye on it when I go through the tutorial.

> It might be better to move the outputs section to the "running computations" hands-on, where they can explore the outputs of the calculations they just ran.

If you know where to place them, please do. I fully agree.

**Comment by ltalirz** [comment]
P.S. Just to provide some context as to the previous structure of the QE example:
The reason we didn't start with running calculations but with the analysis of existing data (not very sexy) was that running an external code was considered more difficult to get started with -- originally there was no `--config` option, and computer setup + code setup was error-prone (we let users go through the interactive prompts); plus there was the whole provenance concept to grasp etc.

Since the basic tutorial from the aiida-core docs now manages to introduce all these concepts in a straightforward manner, there is really no compelling reason anymore to keep this order, and so for the "QE example" I am all for integrating the 'data analysis' into the 'running computations' part wherever possible.

**Comment by ltalirz** [comment]
List of things to check:

 - [ ] Add a comment in the "AiiDA basics" section to mention that it is taken from the basic aiida tutorial (even if just for us to remember in the future)
 - [ ] search for "this tutorial" in the section copied from the docs and replace with the appropriate section name 
 - [ ] replace link to (missing) appendix for computer setup with a link to the aiida-core docs after https://github.com/aiidateam/aiida-core/pull/4216 is merged
 - [x] I added some minor fixes in https://github.com/ltalirz/aiida-tutorials/tree/basics-revamp
 - [x] ~~try to find a place where to mention groups~~ done in https://github.com/aiidateam/aiida-tutorials/pull/231
 - [ ] try to find a place where to show the MC provenance explorer

**Review by ltalirz — APPROVED** [review]
as discussed with @mbercx I think this is good to go - remaining items on the checklist can be handled in a separate PR

File: docs/pages/2020_Intro_Week/sections/basics.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (mbercx) [reply docs/pages/2020_Intro_Week/sections/basics.rst:768]: I added a note to the checklist in #248.

**Comment by mbercx** [comment]
I opened #248 to keep track of the remaining items for the basics section.

**Pull request merged**